scripts/travis-deploy.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```

Comment:
I really like the inventiveness here...butttttt....

## Using .travis.yml to associate branches with environments

@dan and I have used a different approach for associating a branch to an environment. I think it's easy to read - and avoids the need for an intermediate script like this.
Look at [the bibservice's .travis.yml file](https://github.com/NYPL-discovery/bibservice/blob/master/.travis.yml) as an example. Let me know if you like it?

## Documenting which branch deploys to which environment

I've been putting [sections like this](https://github.com/NYPL/scsbuster/#git-workflow--deployment) in the README to document git workflow and how it relates to CI/CD. Would that be helpful in this repo?

Reply:
So I got this alternative method from @nonword . I think the reason he suggested to use the script is for us to deploy to any environment locally without pushing it to remote. So I can deploy to QA for testing without really merging to QA branch.

I think I can add some documentations to .travis.yml so that it will be less discombobulating?

I can also add the branch => environment mapping to README.md

Thanks for the suggestions!

Reply:
On the first point, @nodanaonlyzuul , I'm open to the idea that including a separate `deploy` entry for each of your environments is easier to read. For me the benefits of using a single, central `travis-deploy` script slightly outweighs that. These are those benefits:

1. to DRY,
2. to formalize the association between branch names and deployments (i.e. we have strong cross-repo conventions around what branches deploy where; I want to make that concrete and difficult to mis-type. [Scsbuster seems to take a similar approach fwiw](https://github.com/NYPL/scsbuster/blob/master/provisioning/travis_ci_and_cd/build_and_push_to_ecr.sh))
3. to support `deploy-` scripts that can be run as-is in both Travis and locally

That third point is maybe controversial and kind of depends on whether you view Travis as assistant or gate-keeper. Although I like to use Travis somewhat informally as gate-keeper, I've found it really useful to have `deploy-` scripts that can be run locally. The `--profile` is meaningless to Travis, so either you have to create a separate set of deploy scripts for use by Travis (which is error prone and a lot to read) or you have to do away with local deploying entirely. Although I think we should make it our practice to only ever deploy through Travis, I've found having the ability to quickly deploy to dev/qa in a pinch has been a big help in reducing the amount of time we have to stare at Travis waiting for it to spin up and finish a set of redundant tests just to confirm we got the security group right, for example. Maybe those rare occasions of lost time are worth the longer-term benefits of treating Travis as gate-keeper.

Reply:
If we are ok with this approach,  I will make add some comments on `.travis.yml` for better documentations.

Reply:
Nah - I think it'd would be clear for folks to follow the breadcrumb trail to `deploy.sh`.